### PR TITLE
SI-8364 fixes cxTree lookup for imports

### PR DESCRIPTION
**Pull request by xeno-by**
This is reminiscent of the bug that I recently fixed in paradise:
https://github.com/scalamacros/paradise/commit/0dc4e35883d357b7cbcdfd83b5b4821c1dcc0bb1.

When doing something non-standard with contexts, we usually have to keep
in mind that new contexts are created not only for trees that demarcate
blocks of code, but also for imports.

### CODE REVIEW

**Comment by xeno-by** [comment]
review @retronym

File: src/compiler/scala/tools/nsc/typechecker/Typers.scala [hunk 1]
```diff
@@ -3946,7 +3946,13 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
        *  - simplest solution: have two method calls
        *
        */
-      def mkInvoke(cxTree: Tree, tree: Tree, qual: Tree, name: Name): Option[Tree] = {
+      def mkInvoke(context: Context, tree: Tree, qual: Tree, name: Name): Option[Tree] = {
+        def locateCxTree(context: Context): Tree = {
+          // SI-8364
+          if (context.tree.isInstanceOf[Import]) locateCxTree(context.outer)
+          else context.tree
+        }
```
Comment (retronym) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala]: Let's move this to `Context#enclosingNonImport`

Comment (xeno-by) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala]: Done

**Comment by retronym** [comment]
LGTM

**Pull request merged**